### PR TITLE
Usage composable matchers in specs

### PR DESCRIPTION
**Pull request by Neodelf**
Change matchers implementation according to the [issue#2372](https://github.com/bbatsov/rubocop/issues/2372)

### CODE REVIEW

File: spec/rubocop/options_spec.rb [hunk 1]
```diff
@@ -189,7 +189,7 @@ def abs(path)
 
       it 'requires the passed path' do
         options.parse(['--require', required_file_path, 'example.rb'])
-        expect($stdout.string).to start_with('Hello from required file!')
+        expect($stdout.string).to a_string_starting_with('Hello from required file!')
```
Comment (bbatsov) [inline spec/rubocop/options_spec.rb]: That looks a bit awkward.

Comment (Neodelf) [inline spec/rubocop/options_spec.rb]: What do you mean about awkward?

Comment (bbatsov) [inline spec/rubocop/options_spec.rb]: expect to a_string -> doesn't read well at all

Comment (scottmatthewman) [inline spec/rubocop/options_spec.rb]: I agree with @bbatsov. As the [write-up](http://rspec.info/blog/2014/01/new-in-rspec-3-composable-matchers/#matcher-aliases) suggests, the `a_string_starting_with` method is an alias designed for cases where `start_with` is not grammatically correct, and where the generated failure message would be similarly inelegant.

By changing away from `start_with` here, we'd actually be _introducing_ the sort of hard-to-read grammar that, in different situations, the new aliases are designed to remove.

Comment (Neodelf) [inline spec/rubocop/options_spec.rb]: @scottmatthewman, thanks for explanation. I will change this pull-request

**Comment by bbatsov** [comment]
Make the commit message a bit more descriptive and capitalize the first word in it.

**Comment by Neodelf** [comment]
@bbatsov what do you think about the commit message?

**Comment by bbatsov** [comment]
I'd use the wording "Use contain_exactly rspec matchers where applicable" as it reflects better the scope of the changes.

**Comment by Neodelf** [comment]
No problem, as you say

**Comment by bbatsov** [comment]
👍

**Pull request merged**